### ISSUE TITLE
RATING Tag Issues

### ISSUE THREAD

**Issue by gbreeze** [Borewit/music-metadata#111]
Hello,
some notes...

I think in the typings, the field 'rating' should be an array like:
![image](https://user-images.githubusercontent.com/1906169/42124096-26a452a6-7c5d-11e8-9b10-03a9fd6875ed.png)

Next strange thing is, that I have a mp3 file with a rating of 3.
If I read the file with music-metadata, the value is like 2.500000341.
So i have to do a `const rating = Math.round(oldAudioData.metadata.common.rating[0].rating)`

Thanks, and best regards!

**Comment by Borewit** [comment]
I am not sure what a rating of 3 means, but my best guess is that it is 3 points out of 6.

In `common.rating[0].rating` that would be 2.5 out of 5. Can you drag the file into the issue? I would like to see  where the ugly 0.000000341 is coming from.

Related: #27

**Comment by gbreeze** [comment]
Hey Borewit, 

if I set a rating with 3 (out of 5) in foobar2000 or Mp3Tag, than I get this number.

![image](https://user-images.githubusercontent.com/1906169/42138678-7c70183a-7d81-11e8-9452-2eb07c7dc80b.png)

![image](https://user-images.githubusercontent.com/1906169/42138689-a50d6ea0-7d81-11e8-9134-af271abba729.png)

test file:
[test.zip](https://github.com/Borewit/music-metadata/files/2153188/test.zip)

Thanks!

**Comment by Borewit** [comment]
Hi George,

The track you provided has a POPM field set with a raw value of 128. Where 0 is not rated, 1 worst and 255 is the best rating. So the score is `(128 - 1) / 255 ≈ 0.498`.

The rating in Mp3Tag is defined like this:
> Track rating from 1 = Bad to 5 = Very good like Windows Media Player does it. You can also enter stars * for rating the track. 

That is range of 4 (because 0 is stars is not included).
Number of stars is: 1 + 0.498 * 4 ≈ 1 + 2 ≈ 3 stars.

I guess should correct two things
* [ ]  Add the `-1` in `(128 - 1) / 255 ≈ 0.498`
* [ ]  Change the definition of the [0..5] range, because this is very confusing. Maybe [0..1] to be clearly different then the [1..5] star rating.

**Comment by gbreeze** [comment]
Hey Borewit, big thanks for the explanation!

```
export function convertToSimpleRating(decimalRating: number)
{
    // source
    // https://github.com/Borewit/music-metadata/issues/111#issuecomment-401896457 
    // https://community.mp3tag.de/t/rating-and-popmeter-as-column-in-mp3tag-listview/16082/3    

    // 1 (1)
    if (decimalRating == 0.0196078431372549)
        return 1

    // 2 (64)
    if (decimalRating == 1.2549019607843137)
        return 2

    // 3 (128)
    // 2.5098039215686274

    // 4 (196)
    // 3.843137254901961

    // 5 (255)
    // 5.0

    return Math.round(decimalRating)
}
```

**Comment by Borewit** [comment]
This should to the trick George, **after** I have released the changes:
```` JavaScript
/**
 * Convert to 1-5 star rating
 * @param {number} rating Normalized rating [0..1] (common.rating[n].rating)
 * @returns {number} Number of stars: 1, 2, 3, 4 or 5 stars
 */
export function convertToSimpleRating(rating: number)
{
  return 1 + Math.round(rating * 4);
}
````

**Comment by Borewit** [comment]
Including the 0 starts if a rating is set with an undefined rating value:
```` TypeScript
/**
  * Convert to 1-5 star rating
  * @param {number} rating Normalized rating [0..1] (common.rating[n].rating)
  * @returns {number} Number of stars: 1, 2, 3, 4 or 5 stars
  */
function convertToSimpleRating(rating: number)
{
  return rating === undefined ? 0 : 1 + Math.round(rating * 4);
}

// Unit test the function:
it("should convert to starts", () => {

  assert.equal(convertToSimpleRating(undefined), 0);
  assert.equal(convertToSimpleRating(0), 1);
  assert.equal(convertToSimpleRating(0.1), 1);
  assert.equal(convertToSimpleRating(0.2), 2);
  assert.equal(convertToSimpleRating(0.5), 3);
  assert.equal(convertToSimpleRating(0.75), 4);
  assert.equal(convertToSimpleRating(1), 5);
});
````

**Comment by Borewit** [comment]
Implemented / fixed in [v2.3.0](https://github.com/Borewit/music-metadata/releases/tag/v2.3.0)
